src/query-parser.ts

```diff
@@ -1,18 +1,31 @@
 import { Dictionary } from "@navarik/types"
 import { SearchQuery } from "./types"
 
+const castType = (value, type) => {
+  if (!type){return value}
+  if (type.includes('int') || type.includes('long')) {
```

Comment:
These `ifs` are a perfect case for the strategy pattern (see operator resolution in the query builder)